### PR TITLE
Clang-Format 14

### PR DESCRIPTION
**Pull request by hensldm**
Companion PR to https://github.com/zeldaret/oot/pull/1517

Note I added some `(void*)` casts to the overlay `overlayEntry` computation. This is more likely how they will end up being anyways from #1266 . The format in bigslime is weird, but seems worth it to upgrade anyways.

### CODE REVIEW

**Comment by AngheloAlf** [comment]
Adding the `held` tag until we we have a consensus about the jump to `clang-format-14` and all the Jenkins agents are updated too

**Comment by hensldm** [comment]
Fwiw, I believe all agents have been updated to have both `clang-format-11` and `clang-format-14` for the time being.

**Review by AngheloAlf — APPROVED** [review]
I'm fine with moving to clang-format-14

**Review by jpburnett — APPROVED** [review]

**Comment by hensldm** [comment]
Just want to mention, I had forgotten to update the dockerfile as well as other documentation so went ahead and did that.

**Review by engineer124 — APPROVED** [review]

File: docs/BUILDING_MACOS.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ For macOS, use Homebrew to install the following dependencies:
...
 * bash
-* clang-format 11
+* clang-format-14
 
 You can install them with the following commands:
 
 ```bash
 brew update
-brew install coreutils make python3 libpng bash clang-format@11
+brew install coreutils make python3 libpng bash clang-format@14
```
Comment (AngheloAlf) [inline docs/BUILDING_MACOS.md:20]: Just noticed a problem, the homebrew page for clang-format 14 doesn't seem to exist
https://formulae.brew.sh/formula/clang-format@14

As a reference this is version 11: https://formulae.brew.sh/formula/clang-format@11

**Pull request merged**